src/providers/traits.rs

```diff
@@ -0,0 +1,16 @@
+use crate::evaluation::FlattenedContext;
+
+use super::types::{ProviderMetadata, ResolutionDetails};
+
+pub trait FeatureProvider {
+    fn new() -> Self;
+    fn meta_data(&self) -> ProviderMetadata;
+    fn evaluation<T>(
```

Comment:
I'd recommend we call this `resolve` or `resolution`. We generally refer to the thing the provider does as resolution as opposed to evaluation, which encompasses the entire process of evaluating a flag, including hooks, etc.

https://docs.openfeature.dev/docs/specification/glossary#resolving-flag-values vs https://docs.openfeature.dev/docs/specification/glossary#evaluating-flag-values

Reply:
Okay, I will rename resolution